### PR TITLE
Changes SDF key type again to bytes

### PR DESCRIPTION
**Pull request by jkff**
This should be safe now that the Dataflow worker is properly shaded. Will run ValidatesRunner tests to confirm of course.

### CODE REVIEW

**Comment by jkff** [comment]
Run Dataflow ValidatesRunner

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SplittableParDo.java [hunk 1]
```diff
@@ -391,10 +392,10 @@ public String translateRestrictionCoderId(SdkComponents newComponents) {
    * collection is effectively the same elements as input, but the per-key state and timers are now
    * effectively per-element.
    */
-  private static class RandomUniqueKeyFn<T> implements SerializableFunction<T, String> {
+  private static class RandomUniqueKeyFn<T> implements SerializableFunction<T, byte[]> {
     @Override
-    public String apply(T input) {
-      return UUID.randomUUID().toString();
+    public byte[] apply(T input) {
+      return UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8);
```
Comment (lukecwik) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SplittableParDo.java]: nit: you can convert to bytes directly via https://stackoverflow.com/questions/17893609/convert-uuid-to-byte-that-works-when-using-uuid-nameuuidfrombytesb

**Comment by jkff** [comment]
Run Flink ValidatesRunner

**Comment by lukecwik** [comment]
Ping

**Comment by akedin** [comment]
Closing this as stale, please reopen if incorrect

**Comment by lukecwik** [comment]
@akedin You shouldn't need to close stale requests since they are now closed automatically. I specifically pinged this one because I wanted it to remain open.

**Comment by akedin** [comment]
Ack

**Comment by swegner** [comment]
FYI, I plan on picking this up for @jkff to address feedback and get it merged.

**Comment by swegner** [comment]
Run Flink ValidatesRunner

**Comment by swegner** [comment]
Run Dataflow ValidatesRunner

**Comment by swegner** [comment]
@lukecwik PTAL

**Pull request merged**